Channel position can be changed using a dropdown

gabichelsea opened the pull request:
Trello card link:
[https://trello.com/c/nfo478e3](https://trello.com/c/nfo478e3)

I had to implement only the drop-down tasks.

gabichelsea committed:
Added two drop down list with the default values

gabichelsea committed:
Modified Position and Channel Dto to String in StateHistoryViewRepresentation class

gabichelsea committed:
Dropdowns send the selected value name

gabichelsea committed:
Added findByName method to Position-Channel Repository
Added test cases

gabichelsea committed:
Added getChannelDtoByName method implementation and test

gabichelsea committed:
Removed upperCase test cases from persistence layer

gabichelsea committed:
Added Transactional annotation for getBy..() methods
Added test cases for getPositionDtoByName() method

gabichelsea committed:
Added Channel-Position NotFoundExceptions and message keys

gabichelsea committed:
Added enable/disable for position and channel dropdown

gabichelsea committed:
Added test case when position or channel name is null
Assert dto instead of the name of dto

gabichelsea committed:
Merge branch 'master' into channel-position-can-be-changed-using-a-dropdown

# Conflicts:
#	web/src/main/java/com/epam/rft/atsy/web/StateHistoryViewRepresentation.java
#	web/src/main/webapp/WEB-INF/pages/application_state.jsp

gabichelsea committed:
StatesHistoryEntity modified:
Added positionEntity and channelEntity

Modified StatesHistory schema

gabichelsea committed:
Revert commit

gabichelsea committed:
Modified ApplicationsService and StatesHistoryService

gabichelsea committed:
StateHistoryController:
- Added new state validation

gabichelsea committed:
Before modified test rethinking, rewriting

gabichelsea committed:
Test:
getApplicationDtoById() method

gabichelsea committed:
Test:
- saveOrUpdate() method in ApplicationsService

gabichelsea committed:
Test:
saveApplication() method in ApplicationsService

gabichelsea committed:
Test:
saveStateHistory() method in StatesHistoryService

gabichelsea committed:
Removed Unused Exceptions:
PositionNotFoundException
ChannelNotFoundException

gabichelsea committed:
Modified StateHistoryController tests where it was necessary

gabichelsea committed:
Added test cases for saveOrUpdate() method in StateHistoryController

gabichelsea committed:
Extract literals to constants

gabichelsea committed:
Merge branch 'master' into channel-position-can-be-changed-using-a-dropdown

# Conflicts:
#	service/src/main/java/com/epam/rft/atsy/service/impl/ApplicationsServiceImpl.java
#	web/src/main/resources/i18n/messages_en.properties
#	web/src/main/resources/i18n/messages_hu.properties
#	web/src/test/java/com/epam/rft/atsy/web/controllers/rest/StateHistoryControllerTest.java

lamfalusy commented on persistence/src/main/resources/db/migration/schema/V1_15__AlterStateHistoryTable.sql:
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE states_history ADD COLUMN position_id BIGINT;
```
Be careful with the naming when you merge master to this branch, because others create sql files too with version 15.

lamfalusy commented:
If you click "Felvétel szüneteltetése" state and after then you click "Új jelentkezés", the dropdowns do not work.

lamfalusy commented on web/src/main/webapp/WEB-INF/pages/application_state.jsp:
```diff
@@ -105,47 +104,28 @@
-                                  <option selected disabled>${data.positionName}</option>
+                              <select class="input form-control" name="position" id="position" data-bind="valueWithInit: 'positionName'">
+                                  <option value="${data.positionName}" selected disabled>
+                                      <c:out value="${data.positionName}"/>
```
Missing XSS attack protection.

lamfalusy commented on web/src/main/webapp/WEB-INF/pages/application_state.jsp:
```diff
@@ -105,47 +104,28 @@
-                                  <option selected disabled>${data.channelName}</option>
+                              <select class="input form-control" name="channel" id="channel" data-bind="valueWithInit: 'channelName'">
+                                  <option value="${data.channelName}" selected disabled>
+                                      <c:out value="${data.channelName}"/>
```
Missing XSS attack protection.

gabichelsea committed:
Missing XSS attack protection.

gabichelsea committed:
Added positionEntity and channelEntity to StatesHistoryEntity

gabichelsea committed:
Added getChannelDtoById() method
Added getPositionDtoById() method

gabichelsea committed:
Modified StateHistoryTwoWayConverter, StateHistoryViewTwoWayConverter

gabichelsea committed:
Modified ApplicationStateController
Modified StateHistoryController

gabichelsea committed:
Updated StateHistoryController

gabichelsea committed:
application_state.jsp :
working!

removed unnecessary lines from atsy-statehistory.js

gabichelsea committed:
Added test cases for:
getChannelDtoById() method in ChannelService
getPositionDtoById() method in PositionService

gabichelsea committed:
Removed unused import

gabichelsea committed:
Refactoring StateHistoryTwoWayConverterTest & StateHistoryViewTwoWayConverterTest

gabichelsea committed:
Modified StateHistoryControllerTest
Modified ApplicationStateControllerTest
Modified ApplicationStateController

gabichelsea committed:
Merge branch 'master' into channel-position-can-be-changed-using-a-dropdown

# Conflicts:
#	service/src/main/java/com/epam/rft/atsy/service/PositionService.java
#	service/src/main/java/com/epam/rft/atsy/service/converter/impl/StateHistoryViewTwoWayConverter.java
#	service/src/main/java/com/epam/rft/atsy/service/impl/ApplicationsServiceImpl.java
#	service/src/main/java/com/epam/rft/atsy/service/impl/PositionServiceImpl.java
#	service/src/main/java/com/epam/rft/atsy/service/impl/StatesHistoryServiceImpl.java
#	service/src/test/java/com/epam/rft/atsy/service/converter/StateHistoryViewTwoWayConverterTest.java
#	service/src/test/java/com/epam/rft/atsy/service/impl/ApplicationsServiceImplTest.java
#	service/src/test/java/com/epam/rft/atsy/service/impl/PositionServiceImplTest.java
#	service/src/test/java/com/epam/rft/atsy/service/impl/StatesHistoryServiceImplTest.java
#	web/src/main/webapp/WEB-INF/pages/application_state.jsp

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/controllers/ApplicationStateController.java:
```diff
@@ -56,6 +60,15 @@
+
+  @Autowired
+  private PositionService positionService;
+
```
Delete this duplicated empty line.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/controllers/ApplicationStateController.java:
```diff
@@ -77,16 +90,25 @@ public ModelAndView loadPage(@RequestParam Long applicationId,
             .convert(statesHistoryService.getStateHistoriesByApplicationId(applicationId),
                 StateHistoryViewRepresentation.class);
 
+
```
Delete this duplicated empty line.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/controllers/ApplicationStateController.java:
```diff
@@ -77,16 +90,25 @@ public ModelAndView loadPage(@RequestParam Long applicationId,
-          .build());
+          .build();
+
+      if (clickedStateDTO.getId() == 1) {
```
Write comment about this line, because it seems some kind of magic.

You and I understand what this line means now, but it can be confusing for others or it can be confusing for us later. Another opportunity: use StateDTO's name field if it is unique, it can be more readable.

lamfalusy commented on web/src/main/java/com/epam/rft/atsy/web/controllers/rest/StateHistoryController.java:
```diff
@@ -107,8 +107,14 @@ public ResponseEntity saveOrUpdate(@RequestParam Long applicationId,
       }
 
       if (stateId != null && stateId == 1) {
```
Write comment about this line, because it seems some kind of magic.

The reasons are described in a previous comment.

lamfalusy commented:
Some XSS problem still appear. Create a new position with this name: < script > alert("I am an alert box!"); </ script >. Try to use this new position in state history page.

gabichelsea committed:
Fixed it

gabichelsea committed:
Merge branch 'master' into channel-position-can-be-changed-using-a-dropdown

# Conflicts:
#	web/src/main/resources/i18n/messages_en.properties
